net/redisclient.go

```diff
@@ -226,11 +249,43 @@ func NewRedisRingClient(ro *RedisOptions) *RedisRingClient {
 		r.ring = redis.NewRing(ringOptions)
 		r.log = ro.Log
 		r.metricsPrefix = ro.MetricsPrefix
+
+		if ro.AddrUpdater != nil {
+			if ro.UpdateInterval == 0 {
+				ro.UpdateInterval = defaultUpdateInterval
+			}
+			go func() {
```

Comment:
Lets create a helper function for this

Reply:
why?
I think it's good to copy everything to local func closure and it's a pattern that I follow to have background goroutines started in the NewX func.

Reply:
fixed: I split this out